articles/mongodb-nodejs-custom-ranking/index.md

```diff
@@ -0,0 +1,204 @@
+### How to perform custom ranking for records from a MongoDB database in Node.js
+
+![Hero image](/engineering-education/mongodb-nodejs-custom-ranking/hero.jpg)
+
+#### Introduction
+
+In our daily life problems, we would at one point need to sort our data to give us more insights on how to go about our operations. For example in a school setup, we need to analyze the results of students and give them appropriate positions based on their performance. Due to this, we always have to bear in mind that two or more of them will score the same mark or points. As developers, we have to program the system to give the students the same position and the next student that follows after them an appropriately incremented position. For instance, for these five students and their points, we would proceed as follows:
+
+- Student 1: 21pts - **POS 1**
+- Student 2: 38pts - **POS 2**
+- Student 3: 90pts - **POS 3**
+- Student 4: 90pts - **POS 4**
+- Student 5: 58pts - **POS 5**
+
+After sorting:
+
+- Student 3: 90pts - **POS 1**
+- Student 4: 90pts - **POS 1**
+- Student 5: 58pts - **POS 3**
+- Student 2: 38pts - **POS 4**
+- Student 1: 21pts - **POS 5**
+
+As we can see, the first two students tie in position one and the next student starts from position 3.
+
+One great way that will help us achieve this is using the MongoDB ```sort()``` method. The only issue for us is taking care of the ties and giving them appropriate positions. In this article, we will look at how to use arrays in Node.js to make us attain that from documents(records) from a MongoDB database.
```

Comment:
```sort()``` -> `sort()`

Instead of ` ``` ``` `, you could use ` `` ` for variables, methods or unique names to be highlighted